docs/developer-docs/minimal-example-el7031-motor.md

```diff
@@ -0,0 +1,251 @@
+
+Paste this into your terminal:
+
+```bash
+# Press Enter when prompted
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
+
+sudo apt update
+sudo apt install -y npm nodejs git
+
+git clone git@github.com:qitechgmbh/control.git
+cd control/electron
+npm install
+```
+
+### 4.2 Running the Backend
+
+```bash
+./cargo_run_linux.sh
+```
+
+This script:
+- Builds the backend
+- Grants required system capabilities (raw sockets)
+- Starts EtherCAT communication
+
+Sometimes an error like this appears:
+
+```bash
+....
+  37: <alloc::boxed::Box<F,A> as core::ops::function::FnOnce<Args>>::call_once
+             at /rustc/ed61e7d7e242494fb7057f2657300d9e77bb4fcb/library/alloc/src/boxed.rs:1985:9
+  38: std::sys::thread::unix::Thread::new::thread_start
+             at /rustc/ed61e7d7e242494fb7057f2657300d9e77bb4fcb/library/std/src/sys/thread/unix.rs:126:17
+  39: start_thread
+  40: __clone3
+14:10:46.906  INFO ThreadId(04) server::socketio::init: 167: Socket connected to namespace socket=FXdsVyRVGiN1nLXJ namespace=/main
+```
+
+Re-run the code until the QiTech UI appears.
+
+### 4.3 Running the Frontend
+
+```bash
+# move to the control directory
+cd electron
+npm run start
+```
+
+This launches the QiTech Control dashboard.
+
+---
+
+## 5. Demo
+
+### 5.1 Assigning Devices in the Dashboard
+
+Once the backend + frontend are running, you should see:
+
+![](../assets/UI_start_motor.png)
+
+Make sure that under "Assign" → "Machine Assignment"
+the correct serial number is selected (each device should have the same one, here it is 1).
+Under "Machine" select your new Machine however you named it (I named it "TestMotor V1"), else it will detect that there is something connected to it but the connection won't work.
+
+![](../assets/machine_assigment_motor.png)
+
+**NOW THE MOTOR SHOULD TURN!**
+
+In the interface "TestMotor" on the left side, you can now control the motor (its state and its speed).
+
+![](../assets/UI_motor.png)
+
+---
+
+## 6. Documentation
+
+Use the official documentation of the EL7031-0030 for more information:  
+[Beckhoff EL7031-0030 Documentation](https://download.beckhoff.com/download/document/io/ethercat-terminals/el7031-0030de.pdf)
+
+---
+
+## 7. Software Architecture
+
+### Backend (Rust)
+Located in [machines/src/ethercat_beckhoff/](../../machines/src/ethercat_beckhoff/).
+
+1. **`mod.rs`**: Defines the `MotorTestMachine` struct and holds the state (driver wrapper, enabled state, target velocity).
+2. **`api.rs`**: Handles incoming JSON commands from the frontend (Enable/Disable, Set Velocity) via WebSockets/SocketIO.
+3. **`act.rs`**: The real-time control loop. It updates the `StepperVelocityEL70x1` driver wrapper in every cycle based on the current state.
+4. **`new.rs`**: Initializes the hardware.
+
+### Frontend (TypeScript/React)
+Located in [electron/src/machines/ethercat_beckhoff/](../../electron/src/machines/ethercat_beckhoff/) and [electron/src/routes/routes.tsx](../../electron/src/routes/routes.tsx).
```

Comment:
All three links are dead. I suggest using absolut paths. On github `/` is always points to the repository root